cmd/okctl/handlers/application.go

```diff
@@ -0,0 +1,187 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/AlecAivazis/survey/v2"
+
+	"github.com/oslokommune/okctl/pkg/clients/kubectl/binary"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/oslokommune/okctl/pkg/apis/okctl.io/v1alpha1"
+	"github.com/oslokommune/okctl/pkg/commands"
+	"github.com/oslokommune/okctl/pkg/controller/application/reconciliation"
+	common "github.com/oslokommune/okctl/pkg/controller/common/reconciliation"
+	"github.com/oslokommune/okctl/pkg/okctl"
+	"github.com/oslokommune/okctl/pkg/spinner"
+	"github.com/spf13/cobra"
+)
+
+// HandleApplication knows how to reconcile and purge applications
+func HandleApplication(opts *HandleApplicationOpts) RunEHandler { //nolint:funlen
+	return func(cmd *cobra.Command, _ []string) error {
+		err := opts.Validate()
+		if err != nil {
+			return fmt.Errorf("failed validating options: %w", err)
+		}
+
+		ready, err := reconcileApplicationReadyCheck(opts)
+		if err != nil {
+			return fmt.Errorf("prompting user: %w", err)
+		}
+
+		if !ready {
+			fmt.Fprintln(opts.Okctl.Err, "user wasn't ready to continue, aborting.")
+
+			return nil
+		}
+
+		state := opts.Okctl.StateHandlers(opts.Okctl.StateNodes())
+
+		services, err := opts.Okctl.ClientServices(state)
+		if err != nil {
+			return fmt.Errorf("acquiring client services: %w", err)
+		}
+
+		kubectlClient := binary.New(
```

Comment:
line 48 to 80: to better the readability it could have been something like: `scheduler := createScheduler(opts)` - then the boilerplate of setting up the scheduler doesn't interfere with the execution of the handler.